File: dsa/10+11.graph/104_Maximum_Depth_of_Binary_Tree.java
package graph;


//  * Definition for a binary tree node.
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null){
            return 0;
        }

        int leftMaxDepth = 1 + this.maxDepth(root.left);
        int rightMaxDepth = 1 + this.maxDepth(root.right);

        return Math.max(leftMaxDepth, rightMaxDepth);
    }
}

File: dsa/10+11.graph/104_Maximum_Depth_of_Binary_Tree.py
import typing as t

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def maxDepth(self, root: t.Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        return max(
            1 + self.maxDepth(root.left)
            , 1 + self.maxDepth(root.right)
        )